lib/puppet/type/ntp_config.rb

```diff
@@ -15,6 +15,11 @@
     end
   end
 
+  newproperty(:authenticate) do
```

Comment:
I think we have a thing like `newproperty(:authentication, boolean: true)` would avoid string/bool/symbol issues: https://github.com/puppetlabs/puppet/blob/4.7.0/lib/puppet/property/boolean.rb